### PR TITLE
[#156904185] Fix interest rate scaling issues by bumping charta

### PR DESCRIPTION
**Pull request by nadavhollander**
This PR introduces the following changes:

- Bumps the contracts version to include a fix for the interest rate bug reported by @frostiq
- Fixed tests that were broken by the new contracts
- Misc fix: split collateralized adapter into two files to placate tslint over there being two classes in the same file. 
- Misc fix: updated tslint settings to play nicely with prettier

@saturnial @graemecode @chrismin note that a side-effect of the new contracts version is that we no longer deterministically know that index X will always be associated with token Y -- the order at which the tokens are inserted into the registry during the migrations is non-deterministic, meaning token indices in the registry must be determined dynamically in the tests.

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {
-    "extends": ["tslint:latest"],
+    "extends": [
+        "tslint:latest",
+        "tslint-config-prettier"
```
Comment (graemecode) [inline tslint.json]: not compatible with hound at this point. see https://github.com/houndci/hound/issues/1526

File: __test__/unit/adapters/simple_interest_loan_adapter.spec.ts [hunk 2]
```diff
@@ -521,9 +528,13 @@ describe("Simple Interest Loan Adapter (Unit Tests)", async () => {
 
                 beforeAll(async () => {
                     principalToken = await contracts.loadTokenBySymbolAsync("ZRX", TX_DEFAULTS);
+                    principalTokenIndex = await contracts.getTokenIndexBySymbolAsync("ZRX");
                 });
 
                 it("should return debt order with correctly packed values", async () => {
+                    const encodedIndex = principalTokenIndex.toString().padStart(2, "0");
+                    const termsContractParameters = `0x${encodedIndex}00002a5a058fc295ed0000000000013000c000000000000000000000000000`;
```
[the remark this answers is not in the document]
Comment (nadavhollander) [reply __test__/unit/adapters/simple_interest_loan_adapter.spec.ts]: this is the prettier vs. hound issue @graemecode

File: __test__/unit/adapters/simple_interest_loan_adapter.spec.ts [hunk 3]
```diff
@@ -489,9 +493,13 @@ describe("Simple Interest Loan Adapter (Unit Tests)", async () => {
 
                 beforeAll(async () => {
                     principalToken = await contracts.loadTokenBySymbolAsync("MKR", TX_DEFAULTS);
+                    principalTokenIndex = await contracts.getTokenIndexBySymbolAsync("MKR");
                 });
 
                 it("should return debt order with correctly packed values", async () => {
+                    const encodedIndex = principalTokenIndex.toString().padStart(2, "0");
+                    const termsContractParameters = `0x${encodedIndex}000000000429d069189e000000418c40001000000000000000000000000000`;
```
[the remark this answers is not in the document]
Comment (nadavhollander) [reply __test__/unit/adapters/simple_interest_loan_adapter.spec.ts]: ditto

File: __test__/unit/adapters/simple_interest_loan_adapter.spec.ts [hunk 4]
```diff
@@ -457,9 +458,13 @@ describe("Simple Interest Loan Adapter (Unit Tests)", async () => {
 
                 beforeAll(async () => {
                     principalToken = await contracts.loadTokenBySymbolAsync("REP", TX_DEFAULTS);
+                    principalTokenIndex = await contracts.getTokenIndexBySymbolAsync("REP");
                 });
 
                 it("should return debt order with correctly packed values", async () => {
+                    const encodedIndex = principalTokenIndex.toString().padStart(2, "0");
+                    const termsContractParameters = `0x${encodedIndex}000000000de0b6b3a764000000057820002000000000000000000000000000`;
```
[the remark this answers is not in the document]
Comment (nadavhollander) [reply __test__/unit/adapters/simple_interest_loan_adapter.spec.ts]: ditto

File: __test__/unit/adapters/collateralized_simple_interest_loan_adapter.spec.ts [hunk 5]
```diff
@@ -32,11 +32,11 @@ import { DebtOrder, DebtRegistryEntry } from "src/types";
 // adapters
 import {
     CollateralizedSimpleInterestLoanAdapter,
-    CollateralizedLoanTerms,
     CollateralizerAdapterErrors,
     CollateralizedTermsContractParameters,
     CollateralizedSimpleInterestLoanOrder,
 } from "src/adapters/collateralized_simple_interest_loan_adapter";
+import { CollateralizedLoanTerms } from "src/adapters/collateralized_simple_interest_loan_terms";
```
[the remark this answers is not in the document]
Comment (nadavhollander) [reply __test__/unit/adapters/collateralized_simple_interest_loan_adapter.spec.ts]: @graemecode Where did we land on this?  Is this a hound issue or an actual linting issue?

File: __test__/unit/adapters/collateralized_simple_interest_loan_adapter.spec.ts [hunk 6]
```diff
@@ -277,11 +277,15 @@ describe("Collateralized Simple Interest Loan Adapter (Unit Tests)", () => {
             TX_DEFAULTS,
         );
 
-        const REP_TOKEN = await contracts.loadTokenBySymbolAsync(REP_TOKEN_SYMBOL, TX_DEFAULTS);
-
-        const MKR_TOKEN = await contracts.loadTokenBySymbolAsync(MKR_TOKEN_SYMBOL, TX_DEFAULTS);
+        const tokenSymbols = await Promise.all(
+            Array.from(Array(3).keys()).map((index) =>
```
Comment (graemecode) [inline __test__/unit/adapters/collateralized_simple_interest_loan_adapter.spec.ts]: `Array.from(Array(3).keys())` == `[ 0, 1, 2 ]`?

File: src/adapters/collateralized_simple_interest_loan_terms.ts [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+        }
+    }
+
+    private assertCollateralAmountWithinBounds(collateralAmount: BigNumber) {
+        // Collateral amount cannot be a decimal value.
+        if (TermsContractParameters.isDecimalValue(collateralAmount)) {
+            throw new Error(CollateralizerAdapterErrors.INVALID_DECIMAL_VALUE());
+        }
+
+        if (collateralAmount.lt(0)) {
```
Comment (graemecode) [inline src/adapters/collateralized_simple_interest_loan_terms.ts]: For reference, BigNumber has a `isNegative()` function for readability -- though admittedly it's the same code.

File: __test__/unit/adapters/collateralized_simple_interest_loan_adapter.spec.ts (hunk 6 above)
Comment (nadavhollander) [reply __test__/unit/adapters/collateralized_simple_interest_loan_adapter.spec.ts]: Fair enough :)

**Pull request merged**